linera-sdk/src/service/system_api.rs

```diff
@@ -38,17 +29,15 @@ where
 
 /// Loads the application state (and locks it for writes).
 pub(crate) async fn lock_and_load_view<State: View<ViewStorageContext>>() -> State {
-    let future = wit::Lock::new();
-    future::poll_fn(|_context| -> Poll<Result<(), ViewError>> { future.poll().into() })
-        .await
-        .expect("Failed to lock application state");
+    let promise = wit::Lock::new();
+    promise.wait().expect("Failed to lock application state");
```

Comment:
This seems to be waiting immediately after creating the promise (no poll).

Reply:
Fixed, thanks! There was a minor unexpected issue though, one test called the API using `now_or_never`, which now would return `None`. I fixed it by blocking on the future instead.